sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs

```diff
@@ -945,14 +943,40 @@ private void ValidateCanCreateSenderLink(string entityPath)
             link.Closed += closeHandler;
         }
 
-        private void StopTrackingLinkAsActive(AmqpObject link)
+        private void StopTrackingLinkAsActive(AmqpObject link, Timer authorizationRefreshTimer = null)
         {
+            var activeTimer = default(Timer);
+
             if (link != null)
             {
-                ActiveLinks.TryRemove(link, out var timer);
+                ActiveLinks.TryRemove(link, out activeTimer);
 
-                timer?.Change(Timeout.InfiniteTimeSpan, Timeout.InfiniteTimeSpan);
-                timer?.Dispose();
+                if (activeTimer != null)
+                {
+                    try
+                    {
+                        activeTimer.Change(Timeout.InfiniteTimeSpan, Timeout.InfiniteTimeSpan);
+                        activeTimer.Dispose();
+                    }
+                    catch (ObjectDisposedException)
+                    {
+                    }
+                }
+            }
+
+            // If the refresh timer was created but not associated with the link, then it will need
+            // to be cleaned up.
+
+            if ((authorizationRefreshTimer != null) && (!ReferenceEquals(authorizationRefreshTimer, activeTimer)))
```

Comment:
`StartTrackingLinkAsActive` can throw in some cases, which would cause the refresh timer to not be disposed.